pkg/apis/pingcap/v1alpha1/validation/validation.go

```diff
@@ -42,6 +42,17 @@ func ValidateTidbCluster(tc *v1alpha1.TidbCluster) field.ErrorList {
 	return allErrs
 }
 
+func ValidateTidbMonitor(monitor *v1alpha1.TidbMonitor) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// validate monitor service
+	if monitor.Spec.Grafana != nil {
+		allErrs = append(allErrs, validateService(&monitor.Spec.Grafana.Service, field.NewPath("spec"))...)
+	}
+	allErrs = append(allErrs, validateService(&monitor.Spec.Prometheus.Service, field.NewPath("spec"))...)
```

Comment:
check if monitor.Spec.Prometheus is nil？

Reply:
Prometheus and Reload can't be empty, do you mean need to add some check if these are empty ?
https://github.com/pingcap/tidb-operator/blob/5a47dacda7198eb2c2a426464d6771c2cb0250fd/pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go#L41-L48

Reply:
got it, it is required filed.